src/coreclr/jit/compiler.cpp

```diff
@@ -6363,6 +6390,47 @@ int Compiler::compCompileHelper(CORINFO_MODULE_HANDLE classPtr,
 
     compInitDebuggingInfo();
 
+    // If are an altjit and have patchpoint info, we might need to tweak the frame size
+    // so it's plausible for the altjit architecture.
+    //
+    if (compileFlags->IsSet(JitFlags::JIT_FLAG_OSR))
```

Comment:
```suggestion
    if (!info.compMatchedVM && compileFlags->IsSet(JitFlags::JIT_FLAG_OSR))
```

Shouldn't you also only do this if we know we're running against a mismatched VM (aka, as a cross-altjit)?

Reply:
Yes, thanks for spotting that. My initial version of this was up under where we checked for the mismatch.